atat/src/queues.rs

```diff
@@ -1,47 +1,46 @@
 //! Type definitions for the queues used in this crate.
 
-use heapless::Vec;
 use heapless::{
-    consts,
     spsc::{Consumer, Producer, Queue},
-    ArrayLength,
+    Vec,
 };
 
 pub use crate::error::InternalError;
 pub use crate::Command;
 
 // Queue item types
 pub type ComItem = Command;
-pub type ResItem<BufLen> = Result<Vec<u8, BufLen>, InternalError>;
-pub type UrcItem<BufLen> = Vec<u8, BufLen>;
+pub type ResItem<const BUF_LEN: usize> = Result<Vec<u8, BUF_LEN>, InternalError>;
+pub type UrcItem<const BUF_LEN: usize> = Vec<u8, BUF_LEN>;
 
-pub type ResCapacity = consts::U1;
-pub type ComCapacity = consts::U3;
+pub const RES_CAPACITY: usize = 1;
+pub const COM_CAPACITY: usize = 3;
 
 // Consumers
-pub type ComConsumer = Consumer<'static, ComItem, ComCapacity, u8>;
-pub type ResConsumer<BufLen> = Consumer<'static, ResItem<BufLen>, ResCapacity, u8>;
-pub type UrcConsumer<BufLen, UrcCapacity> = Consumer<'static, UrcItem<BufLen>, UrcCapacity, u8>;
+pub type ComConsumer = Consumer<'static, ComItem, { COM_CAPACITY + 1 }>;
+pub type ResConsumer<const BUF_LEN: usize> =
+    Consumer<'static, ResItem<BUF_LEN>, { RES_CAPACITY + 1 }>;
+pub type UrcConsumer<const BUF_LEN: usize, const URC_CAPACITY: usize> =
+    Consumer<'static, UrcItem<BUF_LEN>, URC_CAPACITY>;
```

Comment:
not sure if this should be `{ URC_CAPACITY + 1}`

Reply:
It should preferrably be, but i can't seem to make it work. 

It comes from the `N - 1` initializer here: https://japaric.github.io/heapless/heapless/spsc/struct.Queue.html